packages/charts/src/hooks/use-tooltip.ts

```diff
@@ -4,7 +4,8 @@ import { TooltipState, DataSelector } from '../types';
 
 export const useTooltip = (
   container: MutableRefObject<any>,
-  computeRelativeToTarget?: boolean
+  computeRelativeToTarget?: boolean,
+  tooltipRef?: MutableRefObject<any>
```

Comment:
IMO name of that argument is a bit misleading - as it do not suggest what behaviour will be achieved by passing tooltip element reference.

I suggest to add before `tooltipRef` - an additional `boolean` argument like: `allowContainerOverflow` (with default `true` value) or some different name. 

Then in condition when `allowContainerOverflow ` equals `false` check that `tooltipRef` is passed if no then throw error.